rfq/order.go

```diff
@@ -260,19 +261,28 @@ func (c *AssetPurchasePolicy) CheckHtlcCompliance(
 			"accepted_quote_id=%v)", htlc, c.AcceptedQuoteId)
 	}
 
+	// Sum the asset balance in the HTLC record.
+	assetAmt, err := htlcRecord.SumAssetBalance(c.AssetSpecifier)
+	if err != nil {
+		return fmt.Errorf("error summing asset balance: %w", err)
+	}
+
 	// Convert the inbound asset amount to millisatoshis and ensure that the
 	// outgoing HTLC amount is not more than the inbound asset amount.
-	assetAmt := rfqmath.NewBigIntFixedPoint(c.AssetAmount, 0)
+	assetAmtFp := rfqmath.BigIntFixedPoint{
```

Comment:
I think the constructor can still be used here?

Reply:
In the latest commits, I've introduced a new fixed-point method, `SetIntValue`, which enables constructing a fixed-point instance from any integer value of the same type as the fixed-point coefficient's type:
```
assetAmtFp := new(rfqmath.BigIntFixedPoint).SetIntValue(assetAmt)
```